aptos-move/block-executor/src/executor.rs

```diff
@@ -1442,16 +1588,17 @@ where
             // Clear by re-initializing the speculative logs.
             init_speculative_logs(signature_verified_block.len());
 
+            // Flush the cache and the environment to re-run from the "clean" state.
+            env.runtime_environment()
+                .flush_struct_name_and_info_caches();
+            CrossBlockModuleCache::flush_at_block_start();
```

Comment:
Yes, I think it is better to have a costly flush but recover here: otherwise if we return panic error because an entry has been cached as non-verified, for instance, we will fail the fallback as well.